### PR TITLE
SAMA5D27 peripheral support - USB Host working

### PR DESCRIPTION
**Pull request by adamfeuer**
### Summary

* SAMA5D27 USB Host support working
* Other peripherals added as a work in progress, but not tested
* Minor nxstyle fixes for some SAMA5D3-Xplained files
* USB Host fixes and testing by @centurysys (Takeyoshi Kikuchi)
* Testing help from @patacongo 

### Impact

* None if you aren't using the SAMA5D27
* Adds USB Host support work for the SAMA5D27
* Adds support for other peripherals but these haven't been tested.

### Limitations / TODO

* Test the other peripherals, especially SPI, I2C, and Ethernet

### Detail

* Much code was copied from the SAMA5D3-Xplained board files. Maybe this should eventually be deduplicated?

### Testing

* Manual with USB flash driver and USB keyboard

### How To Verify

* Configure using the SAMA5D27-XULT defconfig.
* Compile and run
* Plug in a USB flash drive with some files on it
* Do the following in NuttX:

```
NuttShell (NSH) NuttX-8.2
nsh>ls /dev
/dev:
 console
 null
 ramlog
 random
 sda
 ttyS0
 zero
nsh>mkdir /mnt/usb
nsh>mount -t vfat /dev/sda /mnt/usb
nsh>ls /mnt/usb
/mnt/usb:
 foo.txt
 bar.txt
nsh>cat /mnt/usb/bar.txt
0
1
2
3
4
5
6
7
8
9
Hello, World!

```

### CODE REVIEW

File: boards/arm/sama5/sama5d2-xult/configs/nsh/defconfig [hunk 1]
```diff
@@ -26,27 +26,40 @@ CONFIG_AUDIO_NUM_BUFFERS=8
 CONFIG_BOARD_LOOPSPERMSEC=65775
 CONFIG_BOOT_RUNFROMSDRAM=y
 CONFIG_BUILTIN=y
+CONFIG_CDCACM=y
+CONFIG_DEBUG_ERROR=y
+CONFIG_DEBUG_FEATURES=y
+CONFIG_DEBUG_INFO=y
+CONFIG_DEBUG_USB=y
+CONFIG_DEBUG_USB_INFO=y
+CONFIG_DEBUG_WARN=y
```
[the remark this answers is not in the document]
Comment (patacongo) [reply boards/arm/sama5/sama5d2-xult/configs/nsh/defconfig]: Also, any new configurations or any changes to existing configurations need to be documented in the board's README.txt file.

**Comment by patacongo** [comment]
There are quite a few things here.  Perhaps you can "Resolve conversation" when the changes are in place.  Once all of the conversations have been resoved, I will do the merge.

**Comment by adamfeuer** [comment]
@patacongo Thanks— I went through the changed files view and looked to make sure I took care of everything. I resolved all the conversations with your suggestions. It should be good to merge now.

**Comment by liuguo09** [comment]
It seems this PR build breaks sama5d3x-ek:nsh config as below:

It is mainly because of arch/arm/src/sama5/hardware/_sama5d3x_memorymap.h update in this PR.

chip/sam_memories.c: In function 'periphb_physregaddr':
chip/sam_memories.c:86:5: warning: "SAM_PERIPHB_PSECTION" is not defined, evaluates to 0 [-Wundef]
   86 | #if SAM_PERIPHB_PSECTION != SAM_PERIPHB_VSECTION
      |     ^~~~~~~~~~~~~~~~~~~~
chip/sam_memories.c:95:10: error: 'SAM_PERIPHB_PSECTION' undeclared (first use in this function); did you mean 'SAM_PERIPHA_PSECTION'?
   95 |   return SAM_PERIPHB_PSECTION + sectoffset;
      |          ^~~~~~~~~~~~~~~~~~~~
      |          SAM_PERIPHA_PSECTION
chip/sam_memories.c:95:10: note: each undeclared identifier is reported only once for each function it appears in
chip/sam_memories.c: In function 'sysc_physregaddr':
chip/sam_memories.c:149:5: warning: "SAM_SYSC_PSECTION" is not defined, evaluates to 0 [-Wundef]
  149 | #if SAM_SYSC_PSECTION != SAM_SYSC_VSECTION
      |     ^~~~~~~~~~~~~~~~~
chip/sam_memories.c:158:10: error: 'SAM_SYSC_PSECTION' undeclared (first use in this function); did you mean 'SAM_SYSC_VSECTION'?
  158 |   return SAM_SYSC_PSECTION + sectoffset;
      |          ^~~~~~~~~~~~~~~~~
      |          SAM_SYSC_VSECTION
chip/sam_memories.c: In function 'sam_physregaddr':
chip/sam_memories.c:701:45: error: 'SAM_PERIPHA_SIZE' undeclared (first use in this function); did you mean 'SAM_PERIPHB_SIZE'?
  701 |       virtregaddr < (SAM_PERIPHA_VSECTION + SAM_PERIPHA_SIZE))
      |                                             ^~~~~~~~~~~~~~~~
      |                                             SAM_PERIPHB_SIZE
make[1]: *** [sam_memories.o] Error 1
make: *** [arch/arm/src/libarch.a] Error 2
make: *** Waiting for unfinished jobs....
##[error]Process completed with exit code 2.

**Comment by adamfeuer** [comment]
@liuguo09 @patacongo This is my mistake, I am sorry! I'm working on a fix for this file right now.

**Comment by adamfeuer** [comment]
@liuguo09 The fix is in #464, will you check it out and see if it works for you? It builds and runs ok on my SAMA5D3-Xplained board.

**Pull request merged**